### PR TITLE
Import from traits apis

### PR DESCRIPTION
**Pull request by aaronayres35**
With the release of traits 6.2 a section was added to the documentation explicitly requesting users to import from the `.api` modules in traits. See PR https://github.com/enthought/traits/pull/1387

Motivated by this I decided to run through pyface looking for any non api imports from traits.  This PR updates a couple (namely it always imports `UnittestTools` from the `traits.testing.api` and `ETSConfig` from `traitst.etsconfig.api`.

However, I ran into many more examples of traits imports of things not exposed in traits apis.  It is known that this is done but I figured it couldn't hurt to compile a list.  It may be fine to leave them as is, some of these we may want to expose in some traits api, some of them we may not want to be pulling from traits (traits isn't meant to be a storage place for miscellaneous utilities).  This list probably warrants being in a separate or multiple separate issues which I can gladly open eventually.  For now these are the imports I found:

From `traits.trait_base` there are imports of:
 * `xgetattr`
 * `xsetattr`
 * `get_resource_path`
 * `user_name_for`
 * `traits_home`

Additionally we import optional dependencies from `traits.testing.optional_dependencies`

From `traits.trait_list_object` we import `TraitsList` but this is a different `TraitList` from the one exposed in `traits.api`

 from `traits.version` we import `version`

from `traits.util.resource` we import `get_path`

form `traitst.trait_notifiers` we import `set_ui_handler` and `ui_handler`

from `traitss.util.clean_strings` we import `python_name`

and from `traits.util.camel_case` we import `camel_case_to_words`

### CODE REVIEW

**Comment by rahulporuri** [comment]
>  This list probably warrants being in a separate or multiple separate issues which I can gladly open eventually. For now these are the imports I found:

I think this warrants an issue on traits - and we can leave it to the traits maintainer to decide which will be exposed via the api module and which won't.

**Review by rahulporuri — APPROVED** [review]
LGTM

**Comment by aaronayres35** [comment]
> > This list probably warrants being in a separate or multiple separate issues which I can gladly open eventually. For now these are the imports I found:
> 
> I think this warrants an issue on traits - and we can leave it to the traits maintainer to decide which will be exposed via the api module and which won't.

SGTM, I opened https://github.com/enthought/traits/issues/1424

**Pull request merged**
